### ISSUE TITLE
Server with *only* basic auth configured returns WWW-Authenticate "Bearer" whereas it should probably return "Basic"

### ISSUE THREAD

**Issue by immerrr** [jmagnusson/fastapi-security#1]
Hi!

First off, thank you for a nice and useful package! Please find a bug report below.

### Repro
- any fastapi app
-  fastapi security configured with:
```python
from fastapi_security import FastAPISecurity
security = FastAPISecurity()
security.init_basic_auth(settings.BASIC_AUTH_CREDENTIALS)
```
- request:
```
$ curl localhost:8080/api/v1/ -v
```

### Expected Behaviour

Response contains `WWW-Authenticate: Basic` header, as it is the only supported authentication method.

### Actual Behaviour:
```
$ curl localhost:8080/api/v1/ -v
*   Trying 127.0.0.1:8080...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8080 (#0)
> GET /api/v1/ HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.68.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 401 Unauthorized
< date: Thu, 03 Jun 2021 10:34:20 GMT
< server: uvicorn
< www-authenticate: Bearer
< content-length: 43
< content-type: application/json
< 
* Connection #0 to host localhost left intact
{"detail":"Could not validate credentials"}
```

There's a section of code that is responsible for www-authenticate header, https://github.com/jmagnusson/fastapi-security/blob/main/fastapi_security/api.py#L212-L215, that is only returning "Basic" if there is a valid `Authorization: Basic ...` header, which seems like it is not needed. 

According to https://datatracker.ietf.org/doc/html/rfc7235#section-4.1, a WWW-Authenticate header may contain multiple challenges,

![image](https://user-images.githubusercontent.com/579798/120632198-fde1ab00-c468-11eb-92ae-eea711b78233.png),

 so it should be possible to return both Basic and Bearer if they are initialized.

**Comment by jmagnusson** [comment]
Thanks for reporting this @immerrr.

This was fixed in https://github.com/jmagnusson/fastapi-security/commit/f8621bcb39f9bbfc13f5a9437c089bca1a96ff58 and released as [0.4.0](https://pypi.org/project/fastapi-security/0.4.0/)

**Comment by immerrr** [comment]
Great, thank you for such a fast response!
